results/pkg/api/server/v1alpha1/server.go

```diff
@@ -401,6 +419,18 @@ func (s Server) getResultByID(name string) (*pb.Result, error) {
 	return result, nil
 }
 
+func getResultETag(result *pb.Result) (etag string, err error) {
+	tmpEtag := result.GetEtag()
+	result.Etag = ""
+	if b, err := proto.Marshal(result); err == nil {
+		dst := make([]byte, base64.RawURLEncoding.EncodedLen(len(b)))
+		base64.RawURLEncoding.Encode(dst, b)
+		etag = base64.RawURLEncoding.EncodeToString(dst)
+	}
+	result.Etag = tmpEtag
```

Comment:
Instead of having callers set the Etag, we can just set `result.Etag = etag` here.